maintenance/wikia/VideoHandlers/nfcNormalizeVideos.php

```diff
@@ -0,0 +1,130 @@
+<?php
+ini_set('display_errors', 'stderr');
+putenv("SERVER_ID=177");
+
+/**
+ * Class NfcNormalizeVideos
+ * Usage: Use runOnCluster with --method run
+ *
+ * Normalizes video records into NFC form
+ * @see VID-2153 & http://unicode.org/reports/tr15/
+ */
+class NfcNormalizeVideos {
+	protected static $metadataFieldsContainingName = [
+		'title',
+		'name',
+		'keywords',
+	];
+
+	/**
+	 * Used by runOnCluster to Normalize video titles and metadata
+	 *
+	 * @param DatabaseMysql $db
+	 * @param bool $test
+	 * @param bool $verbose
+	 * @param array $params
+	 */
+	public static function run( DatabaseMysql $db, $test, $verbose = false, $params ) {
+
+		if ( !$db->tableExists( 'image' ) || !$db->tableExists( 'video_info' ) ) {
+			echo "ERROR: $params[dbname] (ID:$params[cityId]): image/video_info table not exist.\n";
+			return;
+		}
+
+		foreach ( self::getVideoRows( $db ) as $video ) {
+			$originalName = $video->img_name;
+			if ( \UtfNormal::quickIsNFC( $originalName ) ) {
+				if ( $verbose ) {
+					echo "Already in NFC form; skipping: $originalName\n";
+				}
+				continue;
+			}
+
+			self::updateVideo( $video, $db, $test, $verbose );
+		}
+	}
+
+	/**
+	 * Return an array of videos
+	 *
+	 * @param DatabaseMysql $db
+	 * @return array
+	 */
+	protected static function getVideoRows( DatabaseMysql $db ) {
+		return ( new WikiaSQL() )
+			->SELECT( 'img_name', 'img_metadata', 'img_sha1' )
+			->FROM( 'image' )
+			->WHERE( 'img_media_type' )->EQUAL_TO( 'VIDEO' )
+			->AND_( FluentSql\StaticSQL::RAW( 'img_name <> CONVERT(img_name USING ASCII)' ) )
+			->runLoop( $db, function ( &$videos, $row ) {
+				$videos[] = $row;
+			} );
+	}
+
+	/**
+	 * Update title and other meta data for videos with titles not in normalized NFC form
+	 *
+	 * @param stdClass $video
+	 * @param DatabaseMysql $db
+	 * @param bool $test
+	 * @param bool $verbose
+	 */
+	protected static function updateVideo( $video, DatabaseMysql $db, $test, $verbose ) {
+		$originalName = $video->img_name;
+
+		// image.img_name & video_info.video_title
+		$name = \UtfNormal::toNFC( $originalName );
+		if ( $name == $originalName ) {
+			echo "No change detected. Skipped conversion for: $originalName\n";
+			return;
+		}
+
+		$metadata = self::getNormalizedMetadata( $video );
```

Comment:
@garthwebb now I remember why I did this; metadata would not get `unserialize`d properly, so bunch of things breaks. Here is [an example](http://video.armon.wikia-dev.com/wiki/File:%E3%83%91%E3%82%BA%E3%83%89%E3%83%A9_%E9%AD%94%E6%B3%95%E7%9F%B3) of what I mean. Also here is [an example](http://glee.armon.wikia-dev.com/wiki/File:%E3%82%A6%E3%82%A3%E3%82%AD%E3%82%A2%E3%83%A6%E3%83%8B%E3%83%90%E3%83%BC%E3%82%B7%E3%83%86%E3%82%A3%E3%82%A6%E3%82%A3%E3%82%AD%E3%82%A2%E3%81%AE%E3%83%95%E3%82%A3%E3%83%BC%E3%83%81%E3%83%A3%E3%83%BC%E3%83%84%E3%82%A2%E3%83%BC) when metadata is cleaned.
I am going to add this back in; please see #6095